content/posts/2025-gsoc-clang-doc.md

```diff
@@ -0,0 +1,275 @@
+The YAML backend avoided this problem because its filenames were SymbolIDs, but this meant that the lit tests would have to use regex to find the file for FileCheck.
+Nasty.
+In HTML, you ended up with a buggy HTML page from duplicate tags.
+In JSON, you get a fatal error since two JSON documents can't live in the same file.
+So I used `ItaniumMangleContext` to generate mangled names we could use for the filenames.
+
+### Pull Requests
+- [Serialize record files with mangled name](https://github.com/llvm/llvm-project/pull/148021)
+- [refactor BitcodeReader::readSubBlock](https://github.com/llvm/llvm-project/pull/145835)
+- [refactor JSONGenerator array usage](https://github.com/llvm/llvm-project/pull/145595)
+- [refactor JSON for better Mustache compatibility](https://github.com/llvm/llvm-project/pull/149588)
+
+# Overview
+I implemented a new JSON generator that will serve as the basis for Clang-Doc's documentation generation.
+This will vastly reduce overall lines of code and maintenance burdens.
+I added a lot of tests to increase code coverage and ensure we are serializing all the information necessary for high-quality documentation.
+I refactored our comment handling to streamline the logic that handles them and for better output in the HTML.
+I also explored options for rendering Markdown and began an implemetation for a parser that I plan on working on in the future.
+Along the way, I also did some refactoring to improve code reuse and improved maintenance burdens by reducing boilerplate code.
+
+After my work this summer, Clang-Doc is nearly ready to switch to HTML generation via Mustache templates, which will be a huge milestone.
+It is backed by the JSON generator which will allow for a much more flexible architecture that will change how we generate other documentation formats like our existing Markdown backend.
+All of this was achieved with little to no performance loss.
+I also hope that future contributors have an easier time than I did learning about and working with Clang-Doc.
+The threshold for contributing was high due to a disjointed architecture.
+
+I'm also very excited to present my work and showcase Clang-Doc at this year's LLVM Dev Meeting during a technical talk.
+This is the first time I'll be presenting at a conference and I didn't expect to have the opportunity when I started at the beginning of the summer.
+
+Over the summer, I addressed these issues:
+- [template operator T() produces a bad name](https://github.com/llvm/llvm-project/issues/59812)
+- [Add a JSON backend to clang-doc to better leverage mustache templates](https://github.com/llvm/llvm-project/issues/140094)
+- [Reconsider use of enum InfoType::IT_default](https://github.com/llvm/llvm-project/issues/142888)
+- [Add a JSON backend to clang-doc to better leverage mustache templates](https://github.com/llvm/llvm-project/issues/140094)
```

Comment:
It's not like you shouldn't mention these, but its not really integrated into the post. Maybe some kind of transition would help?